Backend/utils/cronJobs.js

```diff
@@ -1,8 +1,326 @@
 const cron = require('node-cron');
 const { PrismaClient } = require('@prisma/client');
 const prisma = new PrismaClient();
-const fetch = require('node-fetch');
+const { fetchLocationFromMusicBrainz } = require('./musicBrainzUtils');
+const { refreshSpotifyToken } = require('./spotifyUtils');
+const { fetchLocationFromSpotify } = require('./fetchLocationSpotify')
+const { emitUpdate } = require('../socket');
+require('./createSystemUser.js');
 
-cron.schedule('0 * * * *', async () => {
+function delay(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function fetchWithRetry(url, options, retries = 3, delayMs = 1000) {
```

Comment:
this file is quite long - could you go thru and add some comments (maybe above each function or where the code isn't as intuitive) that briefly explains what each block does? this will help us review your code faster as well!

Reply:
Addressed in the "Resolved feedback from PR #11" commit in the "trending-algorithm" branch